### PR TITLE
[WFLY-20059] Create mixed domains tests against WildFly 31

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.redhat.com/browse/WFLY-20059

### CODE REVIEW

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/wfly31/MixedDomainWFLY31TestSuite.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.Suite;
+import org.junit.runners.Suite.SuiteClasses;
+
+/**
+ * Test suite for the standard mixed domain tests.
+ *
+ * @author <a href="kabir.khan@jboss.com">Kabir Khan</a>
+ */
+@Ignore
```
Comment (yersan) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/wfly31/MixedDomainWFLY31TestSuite.java:22]: Ups, I forgot removing this ignore

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/DomainAdjuster.java [hunk 2]
```diff
@@ -272,4 +264,8 @@ private void removeSubsystemExtensionIfExist(DomainClient client, PathAddress su
             // ignored, the extension does not exist
         }
     }
+
+    protected void adjustXpExtensions(DomainClient client, PathAddress profileAddress) throws Exception {
```
Comment (bstansberry) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/DomainAdjuster.java:268]: Call this adjustExpansionExtensions.

XP is a Red Hat product term, and while there are lots of those in this module in this case its simple to not use that.

File: testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDeploymentOverlayTestCase.java [hunk 3]
```diff
@@ -246,7 +246,7 @@ public void testInstallAndOverlayDeploymentOnDC() throws IOException, MgmtOperat
 
         //Remove all CLI style "deployment-overlay remove --name=overlay-test "
         ModelNode cliRemoveOverlay = Operations.createCompositeOperation();
-        cliRemoveOverlay.get(STEPS).add(Operations.createOperation(REMOVE, PathAddress.pathAddress(MAIN_SERVER_GROUP, DEPLOYMENT_OVERLAY_PATH).toModelNode()));
+//        cliRemoveOverlay.get(STEPS).add(Operations.createOperation(REMOVE, PathAddress.pathAddress(MAIN_SERVER_GROUP, DEPLOYMENT_OVERLAY_PATH).toModelNode()));
```
Comment (yersan) [inline testsuite/mixed-domain/src/test/java/org/jboss/as/test/integration/domain/mixed/MixedDeploymentOverlayTestCase.java:249]: This was also an oversight from my side, left from some local tests

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @yersan

**Pull request merged**